vfs/httpfs/union/union.go

```diff
@@ -0,0 +1,128 @@
+// Package union offers a simple http.FileSystem that can unify multiple filesystems at various mount points.
+package union
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/shurcooL/go/vfs/httpfs/vfsutil"
+)
+
+// New creates an union filesystem with the provided mapping of mount points to filesystems.
+//
+// Each mount point must be of form "/mydir". It must start with a '/', and contain a single directory name.
+func New(mapping map[string]http.FileSystem) http.FileSystem {
+	u := &unionFS{
```

Comment:
From a readability standpoint, creating the value and returning the pointer is more explicit. I know by looking at the return I am returning a pointer.

Since you are calling u.bind and bind takes a pointer receiver, you are not saving anything. But if there were other operations on u, where using a pointer was unnecessary, it would also help there.